collectors/cochrane/collector.py

```diff
@@ -0,0 +1,38 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import logging
+import requests
+import tempfile
+import zipfile
+import io
+import shutil
+import os
+from .. import base
+from .parser import parse_record
+logger = logging.getLogger(__name__)
+
+
+def collect(conf, conn, date_from=None, date_to=None):
+    file_count = 0
+    base.helpers.start(conf, 'cochrane', {})
+
+    dirpath = tempfile.mkdtemp()
+    content = requests.get(conf['COCHRANE_ARCHIVE_URL']).content
+    archive = zipfile.ZipFile(io.BytesIO(content))
+    for name in archive.namelist():
+        archive.extract(name, dirpath)
+
+    for filename in os.listdir(dirpath):
+        file_count += 1
+        file_path = os.path.join(dirpath, filename)
+        db_records = parse_record(conf['COCHRANE_ARCHIVE_URL'], file_path)
+        for rec in db_records:
+            rec.write(conf, conn)
+
+    logger.info("Collected %s review files.", file_count)
+    shutil.rmtree(dirpath)
```

Comment:
You don't need to extract the zip files to disk here. This avoids you having to make sure you're removing the files from disk if there's any exception (which you're not doing yet), and makes the code simpler.

You can replace lines 23 to 34 with:

```python
    content = requests.get(conf['COCHRANE_ARCHIVE_URL']).content
    with zipfile.ZipFile(io.BytesIO(content)) as archive:
        for filename in archive.namelist():
            with archive.open(filename, 'rU') as the_file:
                file_count += 1
                db_records = parse_record(conf['COCHRANE_ARCHIVE_URL'], the_file)
                for rec in db_records:
                    rec.write(conf, conn)
```

After these changes, you just need to change `collectors/cochrane/parser.py` renaming the `file_path` parameter to something like `fp` (i.e. file descriptor), and change like 52 to `study_info['file_name'] = fp.name`.